src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -107,10 +118,21 @@ private BindingDataResolver GetResolver(PropertyInfo propInfo, INameResolver nam
                 return GetAppSettingResolver((string)originalValue, appSettingAttr, nameResolver, propInfo);
             }
             // try to resolve with auto resolve ({...}, %...%)
-            if (autoResolveAttr != null && originalValue != null)
+            if (autoResolveAttr != null)
             {
-                _autoResolves[propInfo] = autoResolveAttr;
-                return GetTemplateResolver((string)originalValue, autoResolveAttr, nameResolver, propInfo, contract);
+                if (autoResolveAttr.Default == DefaultResolveValue.MemberName)
+                {
+                    if (originalValue == null || string.IsNullOrWhiteSpace(originalValue.ToString()))
```

Comment:
Seems like these should be part of the if predicate - we only want to do this default resolution if the target doesn't already have a value